src/auto-evo/RunResults.cs

```diff
@@ -118,6 +119,27 @@ public void AddSplitResultForSpecies(Species species, Species splitSpecies, List
             results[splitSpecies].SplitFrom = species;
         }
 
+        public void KillSpeciesInPatch(Species species, Patch patch, bool refundMigrations = false)
+        {
+            AddPopulationResultForSpecies(species, patch, 0);
+
+            // We copy migration list to be able to modify it
+            foreach (var migration in results[species].SpreadToPatches.ToList())
+            {
+                if (results[species].SplitOff != null && results[species].SplitOffPatches?.Contains(patch) == true)
+                    continue;
+
+                if (migration.To == patch)
+                {
+                    results[species].SpreadToPatches.Remove(migration);
+
+                    // We may still penalize the origin patch, the migration would just have died off on its way.
```

Comment:
Please add a TODO comment here saying that it would be nice to leave some trace of this happening so that it can be tracked why the population in this patch was reduced.